modes/ement/evil-collection-ement.el

```diff
@@ -0,0 +1,122 @@
+;;; evil-collection-ement.el --- Evil integration for Ement.el  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2023  Nicholas Vollmer
+
+;; Author:  Nicholas Vollmer
+;; Keywords: convenience
+;; Maintainer: James Nguyen <james@jojojames.com>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.0
+;; Package-Requires: ((emacs "27.1"))
+;; Keywords: evil, emacs, convenience, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Evil bindings for Ement.el
+
+;;; Code:
+(require 'evil-collection)
+(require 'ement nil t)
+(require 'ement-room-list nil t)
+
+(defgroup evil-collection-ement nil
+  "Evil bindings for Ement.el."
+  :group 'evil-collection)
+
+(defcustom evil-collection-ement-want-auto-retro nil
+  "When non-nil, automatically run `ement-room-retro' when scrolled up at `bob'."
+  :type 'boolean)
+
+(defconst evil-collection-ement-maps '(ement-describe-room-mode-map
+                                       ement-room-list-mode-map
+                                       ement-room-mode-map))
+
+(declare-function ement-room-retro "ement-room")
+(defun evil-collection-ement--auto-retro ()
+  "Call `ement-room-retro' scroll requested from beginning of buffer hit."
+  (and (= (line-beginning-position) (point-min))
+       (not (eq evil-state 'emacs))
+       (member this-command '(previous-line evil-scroll-up))
+       (call-interactively #'ement-room-retro)
+       (setq this-command 'ement-room-retro)))
+
+(defun evil-collection-ement--install-auto-retro ()
+  "Install `evil-collection-ement--auto-retro'."
+  (add-hook 'post-command-hook #'evil-collection-ement--auto-retro nil t))
+
+(defun evil-collection-ement-setup ()
+  "Set up `evil' bindings for Ement.el."
+  (evil-collection-define-key 'normal 'ement-describe-room-mode-map
+    (kbd "q") 'quit-window)
+
+  (evil-collection-define-key 'normal 'ement-room-list-mode-map
+    (kbd "d")     'ement-room-list-kill-buffer
+    (kbd "RET")   'ement-room-list-RET
+    (kbd "<return>") 'ement-room-list-RET)
+
+  (evil-collection-define-key '(normal motion) 'ement-room-mode-map
+    (kbd "<")  'ement-room-transient
+    (kbd "<return>")   'ement-room-send-message
+    (kbd "RET")        'ement-rooom-send-message
```

Comment:
@progfolio Just noticed there is a typo here: `rooom`

Reply:
Maybe he wanted that command to send a message faster, like "vroooom"...

Reply:
Time to add it!

Reply:
Ooops! Thank yooou.